### ISSUE TITLE
After updating to 20.5 gocd initiated a start on most of our pipelines

### ISSUE THREAD

**Issue by Preen** [gocd/gocd#8269]
BUG
After updating to 20.5 gocd initiated a start on most of our pipelines which is kind of disturbing.

##### Summary
I upgraded the gocd server and agents, I missed the part with upgrading the DB which I did after upgrade. Then started, logged in and a majority of my pipelines were in start mode. I was able to stop my agents and abort everyone of them...

It looks like its been triggered by github changes, but we havent pushed any changes to 90% of those repos.

##### Environment
Ubuntu 16.04 LTS

![screencapture-gocd-flewid-se-go-pipelines-2020-06-26-10_19_38](https://user-images.githubusercontent.com/364479/85836410-d8e3b300-b796-11ea-8b9b-711e0a6fbca3.png)

**Comment by maheshp** [comment]
@Preen from the screenshot of your dashboard, all the pipelines have an instance counter 1. It looks like your GoCD server is not running with the migrated database but with new H2 db and hence all the pipelines are triggered.
- Did the migration go through fine or did you see any errors?
- Did you replace the old `cruise.h2.db` with the migrated db file `cruise.mv.db`?

If the migration has not be done as per the instrcutions, I would do the following(assuming you are using the default H2 database), 
1. Take a backup and stop the GoCD server.
2. Hope you have a backup of your GoCD server before upgrade. Run the db migration against the backed up database by following the [instructions](https://docs.gocd.org/20.5.0/installation/upgrade_to_gocd_20.5.0.html). 

**I would specifically call out the following steps in instructions to migrate data from H2 => H2**
```
  3. Delete, take a backup of or move away the file /var/lib/go-server/db/h2db/cruise.h2.db.
  4. Replace the old database with the migrated database by moving the file /var/lib/go-server/db/h2db/new_cruise.mv.db to /var/lib/go-server/db/h2db/cruise.mv.db.
  5. Ensure that the file permissions and ownership of the new cruise.mv.db file are correct (same as that of the old cruise.h2.db file).
```
3. Copy over the migrated H2 database file (cruise.mv.db)  to `/var/lib/go-server/db/h2db/` and restart your server.

**Comment by axb21** [comment]
@Preen I am not sure if this is the issue you are experiencing, but in case it is I would recommend triple checking step 4, moving `new_cruise.mv.db` to `cruise.mv.db`. I experienced an issue much like what you describe after the upgrade, and it turned out to be because I had renamed the generated db file to `cruise.mv.db.` (with an extra dot at the end). It took a lot of head-scratching before I noticed; after I fixed the filename everything worked as expected. If this is what happened to you, it looks like go creates a new, mostly empty `cruise.mv.db` file. Make sure to delete or overwrite that one with yours.

Good luck!
